sys/kern/imgact_elf.c

```diff
@@ -1210,9 +1212,17 @@ __CONCAT(exec_, __elfN(imgact))(struct image_params *imgp)
 				goto ret;
 			break;
 		case PT_GNU_STACK:
+#ifdef __ELF_CHERI
+			if (phdr[i].p_flags & PF_X) {
+				uprintf("CheriABI forbids executable stacks\n");
+				error = ENOEXEC;
+				goto ret;
+			}
+#else
```

Comment:
Do we want to set `imgp->stack_prot` or are we just going to require the sysvec's for CheriABI to always be correct? I'm fine with the latter, not sure if we want a comment to that effect.

Reply:
I imagine the latter; only RW_CAP ever makes sense. Arguably we should deprecate `PT_GNU_STACK` entirely but that seems like a bit of a waste of effort.

Reply:
Happy to add a comment that we're using sv_stackprot (I'd also like to fix max_prot for the stack for CheriABI even though it's a no-op in practice).

I suspect we're going to want to reduce the default stack size dramatically as part of colocation and keeping `PT_GNU_STACK` around to let programs that need larger stacks request them might make sense (probably requires more plumbing).